openbb_platform/providers/coingecko/tests/test_coingecko_fetchers.py

```diff
@@ -0,0 +1,253 @@
+"""Tests for CoinGecko fetchers."""
+
+import pytest
+from datetime import date, datetime
+from unittest.mock import Mock, patch
+
+# Skip tests if imports fail (e.g., in CI without provider installed)
+try:
+    from openbb_coingecko.models.crypto_historical import (
+        CoinGeckoCryptoHistoricalFetcher,
+        CoinGeckoCryptoHistoricalQueryParams,
+    )
+    from openbb_coingecko.models.crypto_price import (
+        CoinGeckoCryptoPriceFetcher,
+        CoinGeckoCryptoPriceQueryParams,
+    )
+    from openbb_coingecko.models.crypto_search import (
+        CoinGeckoCryptoSearchFetcher,
+        CoinGeckoCryptoSearchQueryParams,
+    )
+    IMPORTS_AVAILABLE = True
+except ImportError:
+    IMPORTS_AVAILABLE = False
+
+pytestmark = pytest.mark.skipif(
+    not IMPORTS_AVAILABLE, reason="CoinGecko provider not available"
+)
+
+
+class TestCoinGeckoCryptoHistoricalFetcher:
```

Comment:
Please rework this to match the existing provider fetcher unit tests. HTTP interactions are captured as test cassetttes using the pytest_recorder extension. Look at any other provider besides Deribit for samples.